formats/json-tests/commonTest/src/kotlinx/serialization/json/serializers/JsonRawElementTest.kt

```diff
@@ -0,0 +1,187 @@
+package kotlinx.serialization.json.serializers
+
+import kotlinx.serialization.builtins.MapSerializer
+import kotlinx.serialization.json.*
+import kotlinx.serialization.test.assertFailsWithSerialMessage
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class JsonRawElementTest : JsonTestBase() {
```

Comment:
I've expanded the tests here, but I got pretty stuck. Some of the failure messages don't make sense to me.

```
Failed with streaming = STREAMING expected: kotlinx.serialization.json.JsonLiteral<"lorem"> but was: kotlinx.serialization.json.JsonLiteral<"lorem">
```

```
expected:<{ ="invalid key"}> but was:<{" "="invalid key"}>
```

```
Failed with streaming = STREAMING expected: kotlinx.serialization.json.JsonLiteral<[1,2,3]> but was: kotlinx.serialization.json.JsonArray<[1,2,3]>
```

Can someone help out? :)

Reply:
Actually, I think these tests don't have to be checked for decoding, because in most cases they would be different.

For example, if you encode `JsonRawElement("[1,2,3]")`, naturally, it appears in JSON as an array and is parsed back as an array, thus the difference. I don't think we should do anything with that, so it's OK here to test encoding.

Reply:
Thanks, done!